typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -250,9 +251,17 @@
 ;;  make-predicate
 ;;  cast
 
-;; Helper to construct syntax for contract definitions
+;; Helpers to construct syntax for contract definitions
+;; make-contract-def-rhs : Type-Stx Boolean Boolean -> Syntax
 (define (make-contract-def-rhs type flat? maker?)
-  (contract-def-property #'#f `#s(contract-def ,type ,flat? ,maker? untyped)))
+  (define contract-def `#s(contract-def ,type ,flat? ,maker? untyped))
+  (contract-def-property #'#f (λ () contract-def)))
```

Comment:
From https://github.com/racket/typed-racket/pull/365#discussion_r64797153 which is an "outdated diff" for some reason:

@samth :
Let's use promises instead of thunks.

@AlexKnauth :
I would be little bit uncomfortable with that, because even though promises delay computation, it's usually for a different reason than this. Usually when you see a promise, a forcing it works in the same way that evaluating it in the original context would, and mutation isn't "supposed to" change that. In this case, we're delaying the computation specifically so that it can rely on a mutation that will happen later, so promises feel a bit wrong to me.

Does this make sense?